tests/test_terraform.py

```diff
@@ -51,22 +54,66 @@ def test_happy_path(self):
             executor.destroy()
 
 
-class TestDeployClusterUsingCloudQ(unittest.TestCase):
+class BaseClusterTests(ABC, unittest.TestCase):
+    # Tell pytest to skip this test class
+    __test__ = False
+
+    CLUSTER_PASSWORD = f"8_M(p_{str(uuid.uuid4())}"
     QUMULO_CLUSTER_NODES = 4
 
+    vpc_executor_deploy_result: subprocess.CompletedProcess
+    vpc_executor_outputs: Dict[str, str]
+    qumulo_executor_deploy_result: subprocess.CompletedProcess
+    qumulo_executor_outputs: Dict[str, str]
+
     @classmethod
     def setUpClass(cls):
+        if cls is BaseClusterTests:
+            raise unittest.SkipTest("Skip BaseClusterTests base class")
+
+        # Deploy the VPC
         cls.vpc_executor = TerraformExecutor(
             terraform_workspace="test",
             terraform_vars_file="terraform_multi_az.tfvars",
             terraform_vars={"execution_id": str(uuid.uuid4())},
             module_path="tests/vpc-terraform",
             log_level=TerraformLogLevel.INFO,
         )
-        cls.vpc_executor.deploy()
-        cls.outputs = cls.vpc_executor.output()
+        cls.vpc_executor_deploy_result = cls.vpc_executor.deploy()
+        cls.vpc_executor_outputs = cls.vpc_executor.output()
+
+        # Deploy the Qumulo file system
+        cls.qumulo_executor = cls.create_qumulo_terraform_executor()
+        cls.qumulo_executor_deploy_result = cls.qumulo_executor.deploy()
+        cls.qumulo_executor_outputs = cls.qumulo_executor.output()
```

Comment:
Does this mean we'll get multiple VPCs?  One per sub-class?

Reply:
Yes, one per subclass at the moment. When we parallelize the tests soon, we will look into sharing the VPC;.